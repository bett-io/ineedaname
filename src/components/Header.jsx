import React from 'react';
import { Link } from 'react-router-dom';
import { Navbar, NavItem, Nav } from 'react-bootstrap';
import { LinkContainer } from 'react-router-bootstrap';

// The Header creates links that can be used to navigate
// between routes.
const Header = () => (
  <Navbar>
    <Navbar.Header>
      <Navbar.Brand>
        <Link to="/">Home</Link>
      </Navbar.Brand>
    </Navbar.Header>
    <Nav>
      <LinkContainer to="/about">
        <NavItem eventKey={1}>About</NavItem>
      </LinkContainer>
    </Nav>
  </Navbar>
);

export default Header;
