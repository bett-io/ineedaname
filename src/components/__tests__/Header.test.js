/* eslint-disable react/jsx-filename-extension */

import React from 'react';

import toJson from 'enzyme-to-json';
import Header from '../Header';
import { shallow } from 'enzyme';

describe('Header component', () => {
  test('should pass snapshot check', () => {
    const rendered = shallow(<Header />);
    expect(toJson(rendered)).toMatchSnapshot();
  });

  test('contains 1 <NavItem />s', () => {
    expect(shallow(<Header />).find('NavItem')).toHaveLength(1);
  });

  test('contains links to about, repos and hello', () => {
    const component = shallow(<Header />).find('LinkContainer');
    expect(component.findWhere(n => n.props().to === '/about')).toHaveLength(1);
  });
});
