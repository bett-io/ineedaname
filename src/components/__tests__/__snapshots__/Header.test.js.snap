// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should pass snapshot check 1`] = `
<Uncontrolled(Navbar)
  bsStyle="default"
>
  <NavbarHeader>
    <NavbarBrand>
      <Link
        replace={false}
        to="/"
      >
        Home
      </Link>
    </NavbarBrand>
  </NavbarHeader>
  <Nav
    bsClass="nav"
    justified={false}
    pullLeft={false}
    pullRight={false}
    stacked={false}
  >
    <LinkContainer
      activeClassName="active"
      exact={false}
      replace={false}
      strict={false}
      to="/about"
    >
      <NavItem
        active={false}
        disabled={false}
        eventKey={1}
      >
        About
      </NavItem>
    </LinkContainer>
  </Nav>
</Uncontrolled(Navbar)>
`;
